src/app/core/utils/navigation-listener.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { Subject, Subscription } from 'rxjs';
+import { AlertService, HttpError, NavigationNotifierService, NavigationOrigin } from '@hmcts/ccd-case-ui-toolkit';
+
+@Injectable()
+export class NavigationListenerService {
+  public static readonly CASE_CREATED_MSG = 'The case has been created successfully';
+
+  callbackErrorsSubject: Subject<any> = new Subject();
+  navigationSubscription: Subscription;
+
+  constructor(
+    private alertService: AlertService,
+    private navigationNotifier: NavigationNotifierService,
+    private router: Router) {
+  }
+
+  public init() {
```

Comment:
Would it be possible to have an `ngOnInit()` instead of `init()` and whoever takes it in Module class would subscribe automatically? Maybe it is not a best option here though.

Reply:
This is not a component but a service class, so I suspect ngOnInit would fire even if we implemented it. We call init method implicitly in the `CoreComponent's ngOnInit`

Reply:
My bad. Good point